lib/gcloud/datastore/dataset/query_results.rb

```diff
@@ -133,7 +133,7 @@ def initialize arr = []
         #   end
         #
         def next?
-          !no_more?
+          not_finished?
```

Comment:
From the user's point of view, the `next?` method should only designate that they *may* get more results if they run the query. If we always continue the query while `more_results == NOT_FINISHED`, the value of `more_results` the user should only ever see are `MORE_RESULTS_AFTER_LIMIT` or `NO_MORE_RESULTS`. So it seems like if this is the user visible method it should always return `!no_more?` (see my note in the issue about paging vs. batching).